thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -190,18 +189,20 @@ namespace diskann {
     _u64 num_cached_nodes = node_list.size();
     LOG_KNOWHERE_DEBUG_ << "Loading the cache list(" << num_cached_nodes
                         << " points) into memory...";
-    assert(this->nhood_cache_buf == nullptr && "nhoodc_cache_buf is not null");
-    assert(this->coord_cache_buf == nullptr && "coord_cache_buf is not null");
 
     auto ctx = this->reader->get_ctx();
 
-    nhood_cache_buf = new unsigned[num_cached_nodes * (max_degree + 1)];
-    memset(nhood_cache_buf, 0, num_cached_nodes * (max_degree + 1));
+    if (nhood_cache_buf == nullptr) {
+        nhood_cache_buf = new unsigned[num_cached_nodes * (max_degree + 1)];
+        memset(nhood_cache_buf, 0, num_cached_nodes * (max_degree + 1));
```

Comment:
`sizeof(unsigned)` is 4 on my machine, so memset should include `sizeof(unsigned)` as well.
But instead I would just  convert `nhood_cache_buf` from a pointer to a `std::vector<>`.